Documentation for security mitigation (#182)
janus-idp/operator: docs/admin.md

It has to be re-applied to the controller's container after being reconciled by kubernetes processes.

### Recommended Namespace for Operator Installation

> It is recommended to deploy the Backstage Operator in a dedicated default namespace `backstage-system`. This allows cluster administrators to restrict that the operator gets installed on particular namespaces. For instance, the cluster admininistrator can require that the operator to only get installed on `backstage-system` by creating the following `OperatorGroup`:

Reviewer: ```suggestion
It is recommended to deploy the Backstage Operator in a dedicated default namespace `backstage-system`. This allows cluster administrators to restrict that the operator gets installed on particular namespaces. For instance, the cluster administrator can require that the operator only gets installed on `backstage-system` by creating the following [`OperatorGroup`](https://olm.operatorframework.io/docs/concepts/crds/operatorgroup/):
```

```
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: backstage-operator-group
  namespace: backstage-system
spec:
  targetNamespaces:
  - backstage-system
```

> You can choose to deploy the operator in the `openshift-operators` namespace instead. However, you should keep in mind that the Backstage Operator shares the namespace with other operators and therefore any users who can create workloads in that namespace can get their privileges escalated from all operators' service accounts.

Reviewer: ```suggestion
On OpenShift, you can choose to deploy the operator in the `openshift-operators` namespace instead. However, you should keep in mind that the Backstage Operator shares the namespace with other operators, and therefore any users who can create workloads in that namespace can get their privileges escalated from all operators' service accounts.
```

If you plan to deploy to a [restricted environment](https://docs.openshift.com/container-platform/4.14/operators/admin/olm-restricted-networks.html),
you will need to configure your cluster or network to allow these images to be pulled.
For the list of related images deployed by the Operator, see the `RELATED_IMAGE_*` env vars or `relatedImages` section of the [CSV](../bundle/manifests/backstage-operator.clusterserviceversion.yaml).
See also https://docs.openshift.com/container-platform/4.14/operators/admin/olm-restricted-networks.html

#### Custom Backstage Image

> On You can use the Backstage Operator to deploy a backstage application with your custom backstage image by setting the field `spec.application.image` in your Backstage CR. This is at your own risk and it is your responsibility to ensure that the image is from trusted sources, and has been tested and validated for security compliance.

Reviewer: ```suggestion
You can use the Backstage Operator to deploy a backstage application with your custom backstage image by setting the field `spec.application.image` in your Backstage CR. This is at your own risk and it is your responsibility to ensure that the image is from trusted sources, and has been tested and validated for security compliance.
```

If you plan to deploy to a [restricted environment](https://docs.openshift.com/container-platform/4.14/operators/admin/olm-restricted-networks.html),
you will need to configure your cluster or network to allow these images to be pulled.
For the list of related images deployed by the Operator, see the `RELATED_IMAGE_*` env vars or `relatedImages` section of the [CSV](../bundle/manifests/backstage-operator.clusterserviceversion.yaml).
See also https://docs.openshift.com/container-platform/4.14/operators/admin/olm-restricted-networks.html

#### Custom Backstage Image

> You can use the Backstage Operator to deploy a backstage application with your custom backstage image by setting the field `spec.application.image` in your Backstage CR. This is at your own risk and it is your responsibility to ensure that the image is from trusted sources, has been tested and validated for security complinance.

Reviewer: ```suggestion
You can use the Backstage Operator to deploy a backstage application with your custom backstage image by setting the field `spec.application.image` in your Backstage CR. This is at your own risk and it is your responsibility to ensure that the image is from trusted sources, and has been tested and validated for security compliance.
```
